impacket/krb5/ccache.py

```diff
@@ -321,17 +354,25 @@ def __init__(self, data = None):
         self.credentials = []
         self.miniHeader = None
         if data is not None:
-            miniHeader = self.MiniHeader(data)
-            data = data[len(miniHeader.getData()):]
+            ccache_version = data[1]
 
-            headerLen = miniHeader['headerlen']
+            # Only Version 4 contains a header
+            if ccache_version == 4:
```

Comment:
Now that we're obtaining the version and checking it, I'm wondering on whether we shouldn't be raising a `NotImplementedError` for versions 1 and 2 (which are currently failing).